From bf6794eb099a3984dd23d578cabaae9c63b9f9d0 Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Thu, 30 Jun 2022 17:15:14 +0200
Subject: [PATCH] build: Update geocode-glib libsoup3 dependency

geocode-glib 3.26.3 was released today, with optional soup3 support.
When built against soup3, it generates a geocode-glib-2.0 pkg-config
file rather than geocode-glib-1.0.
Adjust our dependency based on initial setup's own soup2 flag.
---
 libgweather/meson.build | 11 ++---------
 1 file changed, 2 insertions(+), 9 deletions(-)

diff --git a/libgweather/meson.build b/libgweather/meson.build
index 08311af4..22738e67 100644
--- a/libgweather/meson.build
+++ b/libgweather/meson.build
@@ -107,19 +107,12 @@ if get_option('soup2')
   libsoup_name = 'libsoup-2.4'
   libsoup_req_version = '>= 2.44.0'
   libsoup_api_version = '2.4'
+  geocode_glib_dep = dependency('geocode-glib-1.0')
 else
   libsoup_name = 'libsoup-3.0'
   libsoup_req_version = '>= 2.99.2'
   libsoup_api_version = '3.0'
-endif
-
-geocode_glib_dep = dependency('geocode-glib-1.0')
-geocode_soup_version = geocode_glib_dep.get_pkgconfig_variable('soupapiversion')
-if geocode_soup_version == ''
-  geocode_soup_version = '2.4' # Pre libsoup3 support
-endif
-if geocode_soup_version != libsoup_api_version
-  error('geocode-glib was built against a different API of libsoup. @0@ instead of @1@.'.format(geocode_soup_version, libsoup_api_version))
+  geocode_glib_dep = dependency('geocode-glib-2.0')
 endif
 
 libsoup_dep = dependency(libsoup_name, version: libsoup_req_version)
-- 
GitLab

